screens/PreviewDirections.js

```diff
@@ -185,23 +231,27 @@ function PreviewDirections (props) {
         }, 100);
     };
     /**
-     * This component will update the preview Map after the user made some selection.
-     * FIXME: A) Not Too Secure, the user can call the fetchMethod non stop once he comes from 
-     * that context, so maybe conditional rendering with the color and disabled it to only allow one
-     * API Call.
+     * This method will fetch the value of the transport Type that is set in the asyncStorange.
+     * By default, the value is driving. 
      */
-    const refreshPreviewMap = () => {
-        if (transportType) {
-            fetchData(transportType);
-            setIsRefreshed(false);
-        }
+    const fetchTransportType = async () => {
+        let name = await AsyncStorage.getItem("thirdCategory");
+        if (name != null)
+        setOutdoorTransportType(name);
     };
-
+    /**
+     * The useEffect is a hook that will constantly (every second) calls fetchTransportType in case teh value of the transport changes
+     * in the PreferenceMenu context. When that happens, the state outdoorTransportType will change which will trigger the fetchData() 
+     * method to be call again. This logic must be implemented in the IndoorMap context since we want to know the other two fields: 
+     * persona and mobility type.
+     */
     useEffect(() => {
```

Comment:
@AlainJobs

Reply:
Because if I don't listen to the transportType in Preference Menu every second, let's say 10seconds, then the map will be refreshed after 10 seconds. I want the map to be refreshed almost instantly!